cmd/process-agent/collector.go

```diff
@@ -576,7 +592,13 @@ func (l *Collector) consumePayloads(results *api.WeightedQueue, fwd forwarder.Fo
 			case checks.Pod.Name():
 				// Orchestrator intake response does not change RT checks enablement or interval
 				updateRTStatus = false
-				responses, err = fwd.SubmitOrchestratorChecks(forwarderPayload, payload.headers, int(orchestrator.K8sPod))
+				// Pod check contains two parts: metadata and manifest.
+				// The manifest payload header has Content-Encoding:zstd allowing us to decompress payload in the intake
+				if payload.headers.Get(headers.ContentEncodingHeader) == headers.ZSTDContentEncoding {
```

Comment:
Similar to above, I think we can make that cleaner through some handlers but for now it should be enough to just move this to a dedicated function